### PR TITLE
Add task

### PR DESCRIPTION
**Pull request by marbetschar**
This Pull Request adds the ability to create a new task. To achieve this, an empty task row is displayed at the top.

Known issue: The new task gets sucessfully created. However, for some reason the task list does not get refreshed. The user needs to navigate to another list and come back to the current one in order to see the newly created task. See the following GIF:

New Task

### CODE REVIEW

File: src/Widgets/TaskRow.vala [hunk 1]
```diff
@@ -122,23 +148,28 @@ public class Tasks.TaskRow : Gtk.ListBoxRow {
...
         var button_box = new Gtk.ButtonBox (Gtk.Orientation.HORIZONTAL);
         button_box.baseline_position = Gtk.BaselinePosition.CENTER;
         button_box.margin_top = 12;
         button_box.spacing = 6;
         button_box.set_layout (Gtk.ButtonBoxStyle.END);
-        button_box.add (delete_button);
+        if (delete_button != null) {
+            button_box.add (delete_button);
+        }
         button_box.set_child_secondary (delete_button, true);
```
Comment (danrabbit) [inline src/Widgets/TaskRow.vala]:
```suggestion
            button_box.add (delete_button);
            button_box.set_child_secondary (delete_button, true);
        }
```

File: src/Widgets/TaskRow.vala [hunk 2]
```diff
@@ -19,12 +19,19 @@
 */
 
 public class Tasks.TaskRow : Gtk.ListBoxRow {
+
+    public enum EditMode {
```
Comment (danrabbit) [inline src/Widgets/TaskRow.vala]: Does this really need to be an enum? It seems like a bool would cover it

File: src/Widgets/TaskRow.vala [hunk 3]
```diff
@@ -122,23 +148,28 @@ public class Tasks.TaskRow : Gtk.ListBoxRow {
...
+            delete_button.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+        }
 
         var cancel_button = new Gtk.Button ();
         cancel_button.label = _("Cancel");
 
         var save_button = new Gtk.Button ();
-        save_button.label = _("Save Changes");
         save_button.get_style_context ().add_class (Gtk.STYLE_CLASS_SUGGESTED_ACTION);
+        save_button.label = _(edit_mode == EditMode.ADD ? "Add Task" : "Save Changes");
```
Comment (danrabbit) [inline src/Widgets/TaskRow.vala]:
```suggestion
        save_button.label = edit_mode == EditMode.ADD ? _("Add Task") : _("Save Changes");
```

File: src/Widgets/TaskRow.vala (hunk 2 above)
Comment (marbetschar) [reply src/Widgets/TaskRow.vala]: Enum is not mandatory, just thought it’s more expressive than a bool.

My personal favorite would be to detect this state based on the ECal.Component itself (some kind of is_new logic). But I did not figure out how.

**Comment by marbetschar** [comment]
Packing the check and icon into a stack is a good idea.

Regarding the other stuff: do you have an idea how we can figure out if a ECal.Component was already added or not? We can then depend the other logic based on that - which makes is a bit more UI agnostic.

**Comment by marbetschar** [comment]
In terms of the task not automatically been shown on the list, that seems to be related tohttps://github.com/elementary/tasks/issues/90 and will be tackled in this dedicated issue.

@danrabbit worked through your feedback and I think the code should be fine now. Can you give this another look so we can get this merged?

**Pull request merged**